spec/04-basic-declarations-and-definitions.md

```diff
@@ -840,6 +840,23 @@ class C extends I {
 Here, it is OK to leave out the result type of `factorial`
 in `C`, even though the method is recursive.
 
+### Tail-Recursive Call Elimination
+
+Method definitions which contain self-recursive invocations in tail position
+are optimized for stack safety. Self-invocations which are the last operation
+before returning from the method are replaced with jumps to the beginning of
+the method, much as in a while loop. Sibling-invocations are also optimized.
+
+This transform is performed automatically, without intervention by the user;
+the user can require it with the annotation, `scala.annotation.tailrec`,
+for cases where deoptimization would likely result in a stack overflow.
+
+```scala
+@tailrec
```

Comment:
precede with `import scala.annotation.tailrec`?

Reply:
Odersky is on the hot seat to improve that, just because it irks you. I loosely said annotations are found in either package.